e2e/ansible/playbooks/resiliency/test-k8s-master-failure/k8s-jenkins-upgrade.yml

```diff
@@ -0,0 +1,275 @@
+          app: jenkins    
+
+     - name: wait for some time to jenkins save changes.
+       wait_for:
+           timeout: 60
+
+ 
+     - name: 6)  Create and Build one sample job inside pod.
+       shell: >
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }} 
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 
+         create-job sample < /var/jenkins_home/template.xml";
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace  }} 
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 build sample" 
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+    
+     - name: 6a)  Running a build job
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 console sample"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}" 
+       register: First_Build
+  
+     - name: 7)  Replace jenkins pod version 
+       replace:
+        path: "{{ result_kube_home.stdout }}/{{ pod_yaml_alias }}"
+        regexp: '{{ item.0 }}'
+        replace: '{{ item.1 }}'
+       delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+       with_together:
+         - "{{replace_item.1}}"
+         - "{{replace_with.1}}"  
+
+     - name: 7a)  Deploy jenkins pod Again
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_task.yml"
+       vars:
+          app_yml: "{{ pod_yaml_alias }}"
+          ns: "{{ namespace }}"
+
+     - name: 7b)  Confirm pod status is running
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+       vars:
+          ns: "{{namespace}}"
+          app: jenkins     
+       
+     - name: 7c)  Verify that the jenkins is a cluster service
+       shell: source ~/.profile; kubectl get svc -n {{ namespace }}
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: result_service_2
+       failed_when: "'jenkins-svc' not in result_service.stdout"
+
+     - name: 8) Get the name of the pod.
+       shell: source ~/.profile; kubectl get pods -n {{ namespace }} | grep jenkins | awk {'print $1'}
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: jenkins_pod_name_2
+
+     - name: Verify jenkins-version post upgrade.
+       shell: source ~/.profile; kubectl describe pod {{jenkins_pod_name_2.stdout }} -n {{ namespace }}  | grep Image | awk {'print $2'} 
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: version
+       failed_when: "'jenkins/jenkins:2.125' not in version.stdout"
+
```

Comment:
@vibhor995  As part of this master failure playbook,there is no task included to bring up/down master. Can you please explain what exactly this test will do? jenkins upgrade or master failure??